### ISSUE TITLE
超参

### ISSUE THREAD

**Issue by nqxytttt** [xuwenxinedu/R3#2]
您好，感谢您的开源。

能否提供CDs这个数据集的超参方便复现？尤其是GRPO部分的参数，我目前使用的参数如下            
--lr 1e-5 --num_generations 8 --sample -1 --num_epochs 1 --info_file $info_file \
--base_model $base_model --output_dir $out --beta 0.01 \
--batch_size 256 --micro_batch_size 8 --lr $lr --seed 42 \
--train_file $train_file --eval_file $eval_file \
--num_iterations 1 --epsilon 0.2 --epsilon_high 0.28 \

**Comment by xuwenxinedu** [comment]
Thanks for your interest.

To reliably reproduce the results in our paper, please first make sure your Python environment is correct. In our tests, even differences in the `transformers` version alone can lead to noticeable discrepancies in results. Below are a few key dependency versions:

- `torch==2.6.0`  
- `transformers==4.51.3`  
- `trl==0.16.1`

After obtaining the SFT model, we recommend first comparing your results with the “only SFT” numbers reported in the ablation study of our paper, and then proceeding to RL training.

For RL training, please also note that we only tested the code on a 2-GPU setup, and we have not verified correctness on setups with more GPUs. The number of GPUs is closely related to the advantage computation, so we strongly recommend running our provided scripts with 2 GPUs.

For the CDs dataset, most of your hyperparameter settings follow our default configuration, which is reasonable. However, after ensuring a 2-GPU environment, I suggest trying a learning rate of `5e-4` or `3e-4`, both of which worked well for me. You can also try tuning additional hyperparameters to obtain the expected results.

Regardless of whether you achieve the expected results, please let us know. You’re welcome to continue the discussion here.

**Comment by xuwenxinedu** [comment]
感谢各位回复！

@cjw-MER 关于需要多少显存才能运行，SFT阶段每张卡大约需要25GB 左右的显存，RL阶段每张卡大约需要15GB左右的显存。

关于论文复现问题，我近期重新在本地对CDs数据集重复了多次实验，得到的结论与我们在论文中展示的结论一致。考虑到代码都是一致的，我个人仍然怀疑是python/conda/cuda环境导致的问题，建议使用我们提供的yaml文件重建一个conda环境来进行实验。

欢迎大家继续讨论！

**Comment by WindSighiii** [comment]
@xuwenxinedu 你好作者，可以补充一下多个latent token的实现吗，我自己写的好像复现不出来多个latent token的结果

**Comment by xuwenxinedu** [comment]
> [@xuwenxinedu](https://github.com/xuwenxinedu) 你好作者，可以补充一下多个latent token的实现吗，我自己写的好像复现不出来多个latent token的结果
好的，不过这个需要等待一段时间，关于2个latent token的具体实现我需要check之前的代码。而这部分代码所在的服务器近期无法登录，等到修复后我会给出具体实现。

**Comment by WindSighiii** [comment]
@xuwenxinedu 非常感谢！！
